percona/controllers/pgc/schedule.go

```diff
@@ -0,0 +1,135 @@
+package pgc
+
+import (
+	"context"
+	"encoding/json"
+	"reflect"
+	"strconv"
+
+	"github.com/percona/percona-postgresql-operator/cmd/pgo-scheduler/scheduler"
+	crv1 "github.com/percona/percona-postgresql-operator/pkg/apis/crunchydata.com/v1"
+	"github.com/pkg/errors"
+
+	v1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type actionType string
+type s struct {
+	job    crv1.CronJob
+	action actionType
+}
+
+var (
+	keep   actionType = "keep"
+	delete actionType = "delete"
+	update actionType = "update"
+	create actionType = "create"
+)
+
+func (c *Controller) handleScheduleBackup(newCluster, oldCluster *crv1.PerconaPGCluster) error {
+	ctx := context.TODO()
+	if newCluster == nil {
+		return nil
+	}
+	if oldCluster != nil && reflect.DeepEqual(newCluster.Spec.Backup.Schedule, oldCluster.Spec.Backup.Schedule) {
+		return nil
+	}
+	if oldCluster == nil {
+		oldCluster = &crv1.PerconaPGCluster{}
+	}
+
+	sm := make(map[string]s)
+	for _, schedule := range oldCluster.Spec.Backup.Schedule {
+		cmName := newCluster.Name + "-" + schedule.Name
+		sm[cmName] = s{schedule, delete}
+	}
+
+	for _, scheduleJob := range newCluster.Spec.Backup.Schedule {
+		cmName := newCluster.Name + "-" + scheduleJob.Name
+		oldSchedule, ok := sm[cmName]
+		sm[cmName] = s{scheduleJob, create}
+		if ok && reflect.DeepEqual(scheduleJob, oldSchedule.job) {
+			sm[cmName] = s{oldSchedule.job, keep}
+		} else if ok {
+			sm[cmName] = s{scheduleJob, update}
+		}
+	}
+
+	for name, s := range sm {
+		switch s.action {
+		case create:
+			scheduleConfigMap, err := getScheduleConfigMap(name, s, newCluster)
+			if err != nil {
+				return errors.Wrap(err, "get schedule config map")
+			}
+			_, err = c.Client.CoreV1().ConfigMaps(newCluster.Namespace).Create(ctx, scheduleConfigMap, metav1.CreateOptions{})
+			if err != nil && !kerrors.IsAlreadyExists(err) {
```

Comment:
i think we should not ignore `ErrAlreadyExists` there, because it should be created, otherwise something is going wrong.